gadgets/fsnotify/program.bpf.c

```diff
@@ -154,7 +154,7 @@ struct {
 	__uint(type, BPF_MAP_TYPE_HASH);
 	__type(key, void *); // struct fsnotify_event *event
 	__type(value, struct enriched_event);
-	__uint(max_entries, 10240);
+	__uint(max_entries, 1024);
```

Comment:
I don't know but, isn't there any way we can profile the gadget with different workloads and verify that the new value is enough in most cases and the old one was too big?

Reply:
@dorser tried with a limit of 128 and could not fill the buffer by looping on `/bin/true`.

Of course it is not a guarantee if it's enough on a different workload or with a slower process processing the inotify events, but 1024 seems fine to me.

Reply:
`fsnotify_insert_event()` uses a [lock](https://github.com/torvalds/linux/blob/6537cfb395f352782918d8ee7b7f10ba2cc3cbf2/fs/notify/notification.c#L93) as well. However, I don't think my benchmark for `exec` applies here as the lock contention is probably different. Specfically for our use case, I'm pretty sure this is enough as we set a small number of watchers on files we expect to see little to no usage.

Reply:
Checking the return value of `bpf_map_update_elem` and using #1882 would be a good way to tell user this map size is too small.